driver/libfuzzer_driver.cpp

```diff
@@ -110,7 +112,35 @@ AbstractLibfuzzerDriver::AbstractLibfuzzerDriver(
   // libFuzzer forwards the command line (e.g. with -jobs or -minimize_crash).
   gflags::ParseCommandLineFlags(&our_argc, &our_argv, false);
 
-  if (std::any_of(argv_start, argv_end, [](const std::string_view &arg) {
+  // Perform modifications of the command line arguments passed to libFuzzer if
+  // necessary by modifying this copy, which will be made the regular argv at
+  // the end of this function.
+  modified_argv = std::vector<char *>(argv_start, argv_end);
+
+  std::string seed;
+  // Search for the last occurence of a "-seed" argument as that is the one that
+  // is used by libFuzzer.
+  auto seed_pos = std::find_if(
+      std::reverse_iterator(argv_end), std::reverse_iterator(argv_start),
+      [](std::string_view arg) { return absl::StartsWith(arg, "-seed="); });
+  if (seed_pos != std::reverse_iterator(argv_start)) {
+    // An explicit seed has been provided on the command-line, record its value
+    // so that it can be forwarded to the agent.
+    seed = absl::StripPrefix(*seed_pos, "-seed=");
+  } else {
+    // No explicit seed has been set. Since Jazzer hooks might still want to use
+    // a seed and we have to ensure that a fuzzing run can be reproduced by
+    // setting the seed printed by libFuzzer, we generate a seed for it here so
+    // that the two stay in sync.
+    unsigned int random_seed = std::random_device()();
+    std::string seed_arg = "-seed=" + std::to_string(random_seed);
+    // This argument can be accessed by libFuzzer at any (later) time and thus
+    // cannot be safely freed by us.
+    modified_argv.push_back(strdup(seed_arg.c_str()));
```

Comment:
@kyakdan This has the side effect of forwarding the `seed=` argument to all children processes in fork mode. I am not sure whether this makes forked fuzzing less efficient. What do you think?

Reply:
As per our discussion, I am now only adding `-seed` to the command line if not in fork/merge/join mode.